runtime/include/chpl-comm-diags.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2004-2018 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _chpl_comm_diags_h_
+#define _chpl_comm_diags_h_
+
+#include "chpl-atomics.h"
+#include "chpl-comm.h"
+
+typedef struct _chpl_atomic_commDiagnostics {
+#define _DIAGS_DECL_ATOMIC(cdv) atomic_uint_least64_t cdv;
```

Comment:
Oops, I missed this until after I'd approved.  Maybe `_COMM_DIAGS_...`?  (I'm imagining a future in which we do something similar with another layer, like tasking.)

Reply:
Ah, to prevent collisions if `_DIAGS_DECL_*` is already set when we're trying to define it  -- changed